contribs/gmf/src/services/themesservice.js

```diff
@@ -103,24 +114,76 @@ gmf.Themes.findTheme_ = function(themes, themeName) {
  * @return {angular.$q.Promise} Promise.
  */
 gmf.Themes.prototype.getBgLayers = function() {
+
   goog.asserts.assert(!goog.isNull(this.promise_));
   return this.promise_.then(goog.bind(
       /**
        * @param {gmf.ThemesResponse} data The "themes" web service response.
-       * @return {Array.<Object>} Array of background layer objects.
+       * @return {angular.$q.Promise}
        */
       function(data) {
-        var bgLayers = data['background_layers'].map(goog.bind(function(item) {
-          goog.asserts.assert('name' in item);
-          goog.asserts.assert('imageType' in item);
+        var promises = data['background_layers'].map(function(item) {
+          var deferred = this.$q_.defer();
+          this.$http_.get(item.url, {
+            cache: false
+          }).then(
+              goog.bind(
+                  function(deferred, item, response) {
```

Comment:
We gonna use https://github.com/camptocamp/ngeo/pull/444/files#diff-4e83f2b133c079468a24cd359e39a6e5R96 in the end.